code/modules/events/brand_intelligence.dm

```diff
@@ -28,30 +37,42 @@
 /datum/round_event/brand_intelligence/announce(fake)
 	var/source = "unknown machine"
 	if(fake)
-		var/obj/machinery/vending/cola/example = /obj/machinery/vending/cola
-		source = initial(example.name)
+		for(var/obj/machinery/vending/vendor in GLOB.machines)
+			if(!is_station_level(vendor.z))
+				continue
+			vendingMachines.Add(vendor)
+		var/obj/machinery/vending/chosen_vendor = pick(vendingMachines)
+		source = chosen_vendor.name
 	else if(originMachine)
 		source = originMachine.name
-	priority_announce("Rampant brand intelligence has been detected aboard [station_name()]. Please stand by. The origin is believed to be \a [source].", "Machine Learning Alert")
+	priority_announce("Rampant brand intelligence has been detected aboard [station_name()]. Please inspect any [source] brand vendors for aggressive marketing tactics, and reboot them if necessary.", "Machine Learning Alert")
 
 /datum/round_event/brand_intelligence/start()
-	for(var/obj/machinery/vending/V in GLOB.machines)
-		if(!is_station_level(V.z))
-			continue
-		vendingMachines.Add(V)
+	var/datum/round_event_control/brand_intelligence/brand_event = control
+	if(brand_event.chosen_vendor)
+		var/chosen_vendor = brand_event.chosen_vendor
+		for(var/obj/machinery/vending/vendor in GLOB.machines)
+			if(!is_station_level(vendor.z) || !istype(vendor, chosen_vendor))
+				continue
+			vendingMachines.Add(vendor)
+	else
+		for(var/obj/machinery/vending/vendor in GLOB.machines)
+			if(!is_station_level(vendor.z))
+				continue
+			vendingMachines.Add(vendor)
```

Comment:
instead of this being an else, it should instead checkfor ``length(vendingMachines)``, that way it will only run if there's no vending machines in the list already (in case admins choose a vending machine that doesn't exist)